fcrepo-http-api/src/main/java/org/fcrepo/http/api/ExternalContentHandler.java

```diff
@@ -163,36 +181,39 @@ private Link parseLinkHeader(final String link) throws ExternalMessageBodyExcept
         return realLink;
     }
 
-    /**
-     * Find the content type for a remote resource
-     * @param url of remote resource
-     * @return the mime-type reported by remote system or "application/octet-stream" if not supplied
-     */
-    private String findContentType(final String url) {
-        if (url == null) {
-            return null;
-        }
+    private void retrieveContentDetails() {
+        final URI uri = getURI();
```

Comment:
It seems like this can potentially return `null`.

Reply:
yeah, I can add in a null check for safety reasons, but i'm pretty sure the validator that gets run before this catches the case of the link not being set.

Reply:
Probably not necessary to add a `null` check. Not that we need to do it here, but it would almost be cleaner to remove some of the superfluous existing `null` checks, like:
- https://github.com/fcrepo4/fcrepo4/blob/9ec28eb345e154f7aac8187d3ea567bc91f39ba7/fcrepo-http-api/src/main/java/org/fcrepo/http/api/ExternalContentHandler.java#L118
- https://github.com/fcrepo4/fcrepo4/blob/9ec28eb345e154f7aac8187d3ea567bc91f39ba7/fcrepo-http-api/src/main/java/org/fcrepo/http/api/ExternalContentHandler.java#L123
- https://github.com/fcrepo4/fcrepo4/blob/9ec28eb345e154f7aac8187d3ea567bc91f39ba7/fcrepo-http-api/src/main/java/org/fcrepo/http/api/ExternalContentHandler.java#L128
- etc